website/content/docs/v0.14/Guides/deploying-cilium.md

```diff
@@ -3,93 +3,175 @@ title: "Deploying Cilium CNI"
+```
+
+However, beware of the fact that the helm generated Cilium manifest contains sensitive key material. As such you should definitely not host this somewhere publicly accessible.
+
+
+## Method 4: Helm manifests inline install
+
+A more secure option would be to include the `helm template` output manifest inside the machine configuration. To do so patch this into your machine configuration:
+
+``` yaml
+inlineManifests:
+    - name: cilium
+      contents: |
+        --
+        # Source: cilium/templates/cilium-agent/serviceaccount.yaml
+        apiVersion: v1
+        kind: ServiceAccount
+        metadata:
+          name: "cilium"
+          namespace: kube-system
+        ---
+        # Source: cilium/templates/cilium-operator/serviceaccount.yaml
+        apiVersion: v1
+        kind: ServiceAccount
+        -> Your cilium.yaml file will be pretty long....
+```
+
+This will install the Cilium manifests at just the right time during bootstrap.
+
+Beware though:
+- Changing the namespace when templating with Helm does not generate a manifest containing the yaml to create that namespace. As the inline manifest is processed from top to bottom make sure to manually put the namespace yaml at the start of the inline manifest.
+- Only add the Cilium inline manifest to the control plane nodes machine configuration.
+- Make sure all control plane nodes have an identical configuration.
+- If you delete any of the generated resources they will be restored whenever a control plane node reboots.
+- As a safety messure Talos only creates missing resources from inline manifests, it never deletes or updates anything.
+- If you need to update a manifest make sure to first edit all control plane machine configurations and then run `talosctl upgrade-k8s` as it will take care of updating inline manifests.
+
+
+## Known issues
+
+- Talos uses the `podSubnets` and `serviceSubnets` settings under the control plane `cluster` config to determine some network settings. Cilium by default (Cluster Scope IPAM) uses its own configuration. Until Talos is more aware of this it would be best to keep these CIDR ranges manually in sync. For more details: [add Node.PodCIDRs to the k8s nodeadress filter](https://github.com/talos-systems/talos/issues/4212)
+
+- Currently there is an interaction between a Kubespan enabled Talos cluster and Cilium that results in the cluster going down during bootstrap after applying the Cilium manifests. For more details: [Kubespan and Cilium compatiblity: etcd is failing](https://github.com/talos-systems/talos/issues/4836)
+
+- When running Cilium with a kube-proxy eBPF replacement (strict mode) there is a conflicting kernel module that results in locked tx queues. This can be fixed by blacklisting aoe_init with extraKernelArgs. For more details: [Cilium on talos "aoe: packet could not be sent on \*. consider increasing tx_queue_len"](https://github.com/talos-systems/talos/issues/4863)
```

Comment:
We've disabled aoe (ref: https://github.com/talos-systems/pkgs/pull/387), not sure if this is a concern anymore.

Reply:
Ah, I wrote this for v0.14.0 but committed the changes to both the 0.14 and 0.15 files. From what version has https://github.com/talos-systems/pkgs/pull/387 been merged into a release?